src/gui/editors/PianoRoll.cpp

```diff
@@ -2782,23 +2748,61 @@ void PianoRoll::paintEvent(QPaintEvent * pe )
 			// update y-pos
 			y -= WHITE_KEY_BIG_HEIGHT;
 		}
-		// label C-keys...
+
+		// Draw the C line in a more prominent color
 		if( static_cast<Keys>( key % KeysPerOctave ) == Key_C )
 		{
-			const QString cLabel = "C" + QString::number( static_cast<int>( key / KeysPerOctave ) );
-			p.setPen( QColor( 240, 240, 240 ) );
-			p.drawText( C_KEY_LABEL_X + 1, y + 14, cLabel );
-			p.setPen( QColor( 0, 0, 0 ) );
-			p.drawText( C_KEY_LABEL_X, y + 13, cLabel );
 			horizCol.setAlpha( 192 );
 		}
 		else
 		{
 			horizCol.setAlpha( 128 );
 		}
+
 		// draw key-line
 		p.setPen( horizCol );
 		p.drawLine( WHITE_KEY_WIDTH, key_line_y, width(), key_line_y );
+
+		// Compute the corrections for the note names
+		int yCorrectionForNoteLabels = 0;
+
+		int keyCode = key % KeysPerOctave;
+		switch (keyCode)
+		{
+		case 0:
+		case 5:
+			yCorrectionForNoteLabels = -4;
+			break;
+		case 2:
+		case 7:
+		case 9:
+			yCorrectionForNoteLabels = -2;
+			break;
+		case 4:
+		case 11:
+			yCorrectionForNoteLabels = 2;
+			break;
+		}
```

Comment:
@AndiEcker Thanks for the review! I was thinking about that as well but then decided to not merge these functions because they are concerned with different things. The function `isWhiteKey` is concerned with determining whether a `key` number corresponds to a note in the C major scale. Therefore it's rather generic and might also be used in other context in the class as well. The commented code block on the other hand is very specific as it is used in the context of rendering the note labels on the keys.

Using something like `yCorrectionForNoteLabels != 0` would be to implicit for my taste and might also break as soon as someone decides to also render the note labels for the black keys which then might have a non zero correction value as well.